### PR TITLE
Combine attributes specified on Page and PageModel

### LINKED ISSUE #11296: Combine attributes specified on a Razor Page with attributes on model
3.0 adds supports for attributes being specified on Razor Page using `@attribute`. But if a model's present, these get ignored. We should instead consider combining attributes.

### PR DESCRIPTION
**Pull request by pranavkm**
Fixes https://github.com/aspnet/AspNetCore/issues/11296

### CODE REVIEW

File: src/Mvc/Mvc.RazorPages/src/ApplicationModels/DefaultPageApplicationModelProvider.cs [hunk 1]
```diff
@@ -98,16 +98,21 @@ public void OnProvidersExecuted(PageApplicationModelProviderContext context)
...
             // Now we want figure out which type is the handler type.
             TypeInfo handlerType;
+            var pageTypeAttributes = pageTypeInfo.GetCustomAttributes(inherit: true);
+            object[] handlerTypeAttributes;
             if (modelProperty.PropertyType.IsDefined(typeof(PageModelAttribute), inherit: true))
             {
                 handlerType = modelTypeInfo;
+
+                // If a PageModel is specified, combine the attributes specified on the Page and the Model type.
+                handlerTypeAttributes = modelTypeInfo.GetCustomAttributes(inherit: true).Concat(pageTypeAttributes).ToArray();
```
Comment (rynowak) [inline src/Mvc/Mvc.RazorPages/src/ApplicationModels/DefaultPageApplicationModelProvider.cs]: Does it make sense to have the model type attributes first or the page type first? I can't really think of a good way to decide this strongly.

Comment (pranavkm) [reply src/Mvc/Mvc.RazorPages/src/ApplicationModels/DefaultPageApplicationModelProvider.cs]: I wanted the model (code) to be treated as being more significant than the page (code-gen). It also keeps the semantics closer to 2.x if you happened to have attributes on a custom Page base type. 

As a user, we really would recommend being explicit if the ordering is significant.

Comment (rynowak) [reply src/Mvc/Mvc.RazorPages/src/ApplicationModels/DefaultPageApplicationModelProvider.cs]: Makes sense, would be nice to have a comment.

File: src/Mvc/test/Mvc.FunctionalTests/RazorPagesWithBasePathTest.cs [hunk 2]
```diff
@@ -695,6 +695,40 @@ public async Task RoundTrippingFormFileInputWorks()
             await response.AssertStatusCodeAsync(HttpStatusCode.OK);
         }
 
+        [Fact]
+        public async Task AuthAttribute_AppliedOnPageWorks()
```
Comment (rynowak) [inline src/Mvc/test/Mvc.FunctionalTests/RazorPagesWithBasePathTest.cs]: @DamianEdwards

**Review by rynowak — APPROVED** [review]

**Comment by pranavkm** [comment]
@aspnet-hello triage https://dev.azure.com/dnceng/public/_build/results?buildId=272421&view=logs

**Comment by aspnet-hello** [comment]
This comment was made automatically. If there is a problem contact aspnetcore-build@microsoft.com.

I've triaged the above build. I've created/commented on the following issue(s)
https://github.com/aspnet/AspNetCore-Internal/issues/2839
https://github.com/aspnet/AspNetCore-Internal/issues/2843
https://github.com/aspnet/AspNetCore-Internal/issues/2711

**Pull request merged**